src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList(int capacity) {
+        if (capacity <= 0) {
+            throw new IllegalArgumentException("Array`s capacity cannot be <= 0");
+        } else {
+            array = (T[]) new Object[capacity];
+            this.size = 0;
```

Comment:
remove this line, but remember [where](stackoverflow.com/a/2411283) we should use `this`